app/models/miq_vim_broker_worker.rb

```diff
@@ -19,9 +19,12 @@ class MiqVimBrokerWorker < MiqWorker
     return 1
   }
 
-  def self.has_required_role?
-    return false if emses_to_monitor.empty?
-    super
+  def self.should_start_worker?
+    super && emses_to_monitor?
```

Comment:
This is the main driver of this PR - want to get rid of the call to `emses_to_monitor` all the time